test/generators/resource_generator_test.rb

```diff
@@ -0,0 +1,77 @@
+require "test_helper"
+require "generators/godmin/resource/resource_generator"
+
+module Godmin
+  class ResourceGeneratorTest < ::Rails::Generators::TestCase
+    tests ResourceGenerator
+    destination File.expand_path("../../tmp", __FILE__)
+    setup :prepare_destination
+
+    def test_resource_generator_in_standalone_install
+      system "cd #{destination_root} && rails new . --skip-test --skip-spring --skip-bundle --skip-git --quiet"
+      system "cd #{destination_root} && bin/rails generate godmin:install --quiet"
+      system "cd #{destination_root} && bin/rails generate godmin:resource foo bar --quiet"
+
+      assert_file "config/routes.rb", /resources :foos/
+      assert_file "app/views/shared/_navigation.html.erb", /<%= navbar_item Foo %>/
+
+      assert_file "app/controllers/foos_controller.rb" do |content|
+        expected_content = <<-CONTENT.strip_heredoc
+          class FoosController < ApplicationController
+            include Godmin::Resources::ResourceController
+          end
+        CONTENT
+        assert_match expected_content, content
+      end
+
+      assert_file "app/services/foo_service.rb" do |content|
+        expected_content = <<-CONTENT.strip_heredoc
+          class FooService
+            include Godmin::Resources::ResourceService
+
+            attrs_for_index :bar
+            attrs_for_show :bar
+            attrs_for_form :bar
+          end
+        CONTENT
+        assert_match expected_content, content
+      end
+    end
+
+    def test_resource_generator_in_engine_install
+      system "cd #{destination_root} && rails new . --skip-test --skip-spring --skip-bundle --skip-git --quiet"
+      system "cd #{destination_root} && bin/rails plugin new fakemin --mountable --quiet"
+      system "cd #{destination_root} && fakemin/bin/rails generate godmin:install --quiet"
+      system "cd #{destination_root} && fakemin/bin/rails generate godmin:resource foo bar --quiet"
```

Comment:
There is a `run_generator` command provided by `Rails::Generators::TestCase`, however, that will run the generator from the `destination_root`, and we want to run it from within the engine, so I went with `system` instead.